ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcServerProtocolClient.java

```diff
@@ -36,23 +36,44 @@
 import org.slf4j.LoggerFactory;
 
 import java.io.Closeable;
+import java.util.concurrent.CompletableFuture;
 
 /**
  * This is a RaftClient implementation that supports streaming data to the raft
  * ring. The stream implementation utilizes gRPC.
  */
 public class GrpcServerProtocolClient implements Closeable {
+  // Common channel
   private final ManagedChannel channel;
-  private final TimeDuration requestTimeoutDuration;
-  private final RaftServerProtocolServiceBlockingStub blockingStub;
+  // Channel and stub for heartbeat
+  private ManagedChannel hbChannel;
+  private RaftServerProtocolServiceStub hbAsyncStub;
```

Comment:
Change them to final and set them to null when they are disabled.  Then, we don't need another useSeparateHBChannel field.